rpc/auth_test.go

```diff
@@ -137,6 +137,53 @@ func TestMakeSimpleMultiAuthHandler(t *testing.T) {
 	})
 }
 
+func TestMakeSimpleMultiAuthPairHandler(t *testing.T) {
+	test.That(t, func() {
+		MakeSimpleMultiAuthPairHandler([]map[string]string{{"one": "1"}}, nil)
+	}, test.ShouldPanicWith, "expected at least one payload")
+
+	t.Run("with no entities should always fail", func(t *testing.T) {
+		handler := MakeSimpleMultiAuthPairHandler(nil, []map[string]string{{"something": "nothing"}})
+		_, err := handler.Authenticate(context.Background(), "", "something")
+		test.That(t, err, test.ShouldNotBeNil)
+		_, err = handler.Authenticate(context.Background(), "entity", "something")
+		test.That(t, err, test.ShouldNotBeNil)
+	})
+
+	t.Run("should validate entities and (keyID, key) mappings", func(t *testing.T) {
+		expectedEntitiesMap := []map[string]string{{"one": "1"}, {"two": "2"}, {"three": "3"}}
```

Comment:
This might not need to be a map as it looks like you're discarding the values anyways.